### PR TITLE
Update to Tokio 0 3

### PR DESCRIPTION
**Pull request by messense**

### CODE REVIEW

File: src/fsm.rs [hunk 1]
```diff
@@ -69,22 +58,15 @@ impl<AF: AddressFamily> FSM<AF> {
...
+        let mut buf = tokio::io::ReadBuf::new(&mut recv_buf);
         loop {
-            let (bytes, addr) = match self.socket.poll_recv_from(cx, &mut buf) {
-                Poll::Ready(Ok((bytes, addr))) => (bytes, addr),
+            let addr = match self.socket.poll_recv_from(cx, &mut buf) {
+                Poll::Ready(Ok(addr)) => addr,
                 Poll::Ready(Err(err)) => return Err(err),
                 Poll::Pending => break,
             };
-            // Is moot for certain platforms (Windows will throw a <10040> error from poll_recv)
```
Comment (messense) [inline src/fsm.rs:79]: I am not sure about this, but it seems that there is no way to do this is Tokio 0.3?

Comment (willstott101) [reply src/fsm.rs:79]: Yeah looks like this was probably cleared up in Tokio - which is nice.

**Review by willstott101 — COMMENTED** [review]
Generally looks good to me. What's driven the change from `net2` to `socket2`?

**Comment by willstott101** [comment]
Never-mind I see it's deprecated. I'll go ahead and merge and release this later.

**Comment by willstott101** [comment]
Release as 0.5.0 - new minor version due to the tokio bump

**Pull request merged**